symbolic-symcache/src/new/raw.rs

```diff
@@ -10,7 +10,7 @@ const SYMCACHE_MAGIC_BYTES: [u8; 4] = *b"SYMC";
 /// The magic file preamble to identify SymCache files.
 ///
 /// Serialized as ASCII "SYMC" on little-endian (x64) systems.
-pub const SYMCACHE_MAGIC: u32 = u32::from_be_bytes(SYMCACHE_MAGIC_BYTES);
+pub const SYMCACHE_MAGIC: u32 = u32::from_le_bytes(SYMCACHE_MAGIC_BYTES);
```

Comment:
this change here is wrong and would result in a byte flip.
We want this magic constant to be serialized as `"SYMC"` on x64 machines. Maybe we can even assert that in a test:
`buf.starts_with(b"SYMC")`

BTW, there is a `cfg()` attribute that checks for endianness which we should use in that case.

Maybe have @jan-auer double-check this, since we had a couple of discussions regarding endianness previously. The existing old symcache is not endian-aware for that matter. It would fail at the version check in that case.

Reply:
I changed this because some tests were failing due to it. Something weird's going on.

Reply:
I added the `buf.starts_with(b"SYMC")` assertion to numerous tests and they all succeed. I think this might have just been wrong before.